### PR TITLE
Return `nan` when one variable to be correlated has zero variance in JIT GroupBy Apply

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Closes https://github.com/rapidsai/cudf/issues/13875

### CODE REVIEW

**Comment by brandon-b-miller** [comment]
This was found during https://github.com/rapidsai/cudf/pull/13813

File: python/cudf/udf_cpp/shim.cu [hunk 1]
```diff
@@ -643,9 +643,8 @@ __device__ double BlockCorr(T* const lhs_ptr, T* const rhs_ptr, int64_t size)
 {
   auto numerator   = BlockCoVar(lhs_ptr, rhs_ptr, size);
   auto denominator = BlockStd(lhs_ptr, size) * BlockStd<T>(rhs_ptr, size);
-
   if (denominator == 0.0) {
-    return 0.0;
+    return std::numeric_limits<double>::quiet_NaN();
```
Comment (galipremsagar) [inline python/cudf/udf_cpp/shim.cu:647]: Can we have a pytest added to test this?

**Review by vyasr — APPROVED** [review]
Approving assuming that a new test is added as per Prem's request.

File: python/cudf/udf_cpp/shim.cu (hunk 1 above)
Comment (brandon-b-miller) [reply python/cudf/udf_cpp/shim.cu:647]: Added a simple test in https://github.com/rapidsai/cudf/pull/13884/commits/26ad24a7530563f553bbff3c297e4794aa3fee53. More tests pending, including one which covers this case, at https://github.com/rapidsai/cudf/pull/13854.

**Review by galipremsagar — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Comment by brandon-b-miller** [comment]
@galipremsagar jinx :)

**Pull request merged**